tests/publiccloud/check_registercloudguest.pm

```diff
@@ -126,14 +126,14 @@ sub check_instance_registered {
         record_info('zypper lr', $instance->ssh_script_output(cmd => 'zypper -t lr ||:'));
         die('The list of zypper repositories is empty.');
     }
-    if ($instance->ssh_script_output(cmd => 'sudo ls /etc/zypp/credentials.d/ | wc -l') == 0) {
+    if ($instance->ssh_script_output(cmd => 'sudo ls -A /etc/zypp/credentials.d/') != 0) {
```

Comment:
The orginal command returned stdout, therefore it checked the file count == 0: your change will always return 0.

Reply:
Wasn't 100% sure how `ssh_script_output(cmd =>` handled it, but I think that just having the check done in bash - i.e. `[ $(ls -A /etc/zypp/credentials.d/) ]` and `[ -z $(ls -A /etc/zypp/credentials.d/) ]` respectively, is better. I chickened out last minute and removed the logic with the last commit

Reply:
Since we just need to know if the folder is empty or not and we don't actually care about how many files there are.

Reply:
1. ls without  `-A` also not return `.` , `..` so why do we need this flag ?
2. I would agree with Maurizio here , counting lines and compare **count** with 0 looks more solid then your version when we can get empty list or some list which we would compare with 0 . Your version might still work but it is just another way to do same thing not something what I can call an **improvement** . Can you elaborate why do you think that this way is better than what we have in master now ?

Reply:
1. Because without `-A` it will always return not-empty exactly because of `.` and `..`.
2. Because we want to check if a folder is empty. In my mind it is an improvement because it's simpler - 1 command and we take the exit code instead of passing the output and then counting it.

Reply:
> 1. Because without `-A` it will always return not-empty exactly because of `.` and `..`.

no it is not . 

```
asmorodskyi:~/ $ mkdir /tmp/empty                                                                                                                                                    
asmorodskyi:~/ $ ls /tmp/empty                                                                                                                                                       
asmorodskyi:~/ $ 
```

I assume you confusing `ls` with `ls -a`  , also I recommend you to double-check your aliases many distros doing nasty things by overwriting default commands behavior , for example in my case  I had such thingy : 
```
alias | grep "^ls="                                                                                                                                              
ls='ls --color=tty'
```
but AFAIK this should not influence display of `.` and `..`

> 2. Because we want to check if a folder is empty. In my mind it is an improvement because it's simpler - 1 command and we take the exit code instead of passing the output and then counting it.

but `script_output` not returning exit code but stdout , you need to switch to `script_run` to get exit code

Reply:
1. `ls` by itself will not show `.` dotfiles. I do not know if we want to see those or care about them, but if they exist - the folder is not empty. `ls -a` will show the dotfiles but also `.` `..`. So the only way to be certain by using `ls` would be `ls -A`. 
2. I did not realize that, good catch!

Reply:
> ls by itself will not show . dotfiles. I do not know if we want to see those or care about them, but if they exist - the folder is not empty. ls -a will show the dotfiles but also . ... So the only way to be certain by using ls would be ls -A.

ok , make sense